src/rpc/rawtransaction.cpp

```diff
@@ -229,29 +239,36 @@ UniValue gettxoutproof(const JSONRPCRequest& request)
        oneTxid = hash;
     }
 
-    LOCK(cs_main);
-
     CBlockIndex* pblockindex = nullptr;
-
     uint256 hashBlock;
-    if (!request.params[1].isNull())
     {
-        hashBlock = uint256S(request.params[1].get_str());
-        pblockindex = LookupBlockIndex(hashBlock);
-        if (!pblockindex) {
-            throw JSONRPCError(RPC_INVALID_ADDRESS_OR_KEY, "Block not found");
-        }
-    } else {
-        // Loop through txids and try to find which block they're in. Exit loop once a block is found.
-        for (const auto& tx : setTxids) {
-            const Coin& coin = AccessByTxid(*pcoinsTip, tx);
-            if (!coin.IsSpent()) {
-                pblockindex = chainActive[coin.nHeight];
-                break;
+        LOCK(cs_main);
```

Comment:
Instead of nesting, repeat the lock:
```cpp
if (!request.params[1].isNull()) {
    hashBlock = uint256S(request.params[1].get_str());
    LOCK(cs_main);
    ...
} else {
    LOCK(cs_main);
    ...
}
```